### PR TITLE
feat: edit community channels

### LINKED ISSUE #2235: feat: Edit community channel, add description and private fields
# Problem

Community channels are currently not editable.

# Implementation

Community channel fields should be editable. A field for `private` and `description` should also be added in the response object to support the UI.

# Acceptance Criteria

All current fields in the community channel (name, description, and private) should be editable.

# Future Steps

For now, `private` field should be a boolean to support a UI switch/toggle. In the future, this should be updated to accept permissions.

### PR DESCRIPTION
**Pull request by emizzle**
Closes #2235.

Add ability to edit community channels including name, description, private (bool).

`Description` and `Private` fields were added to the `Chat` object and the `Description` field was added to the `CommunityChat` object.

`private` is a boolean value to support true/false UI switches, however in the future it will need to change to support permissions.

A short summary which serves as a squashed-commit message.

A description to understand introduced changes without reading the code.

Important changes:
- [x] Something worth noting for reviewers.

Closes #

### CODE REVIEW

File: protocol/communities/community.go [hunk 1]
```diff
@@ -242,6 +245,40 @@ func (o *Community) CreateChat(chatID string, chat *protobuf.CommunityChat) (*Co
 	return changes, nil
 }
 
+func (o *Community) EditChat(chatID string, chat *protobuf.CommunityChat) (*CommunityChanges, error) {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+
+	if o.config.PrivateKey == nil {
```
Comment (vkjr) [inline protocol/communities/community.go]: I'd suggest to use `!o.IsAdmin()` for better readability

File: protocol/communities/community.go [hunk 2]
```diff
@@ -242,6 +245,40 @@ func (o *Community) CreateChat(chatID string, chat *protobuf.CommunityChat) (*Co
...
+	if _, exists := o.config.CommunityDescription.Chats[chatID]; !exists {
+		return nil, ErrChatNotFound
+	}
+
+	o.config.CommunityDescription.Chats[chatID] = chat
+
+	o.increaseClock()
+	
+	changes := o.emptyCommunityChanges()
+	if changes.ChatsModified[chatID] == nil {
```
Comment (vkjr) [inline protocol/communities/community.go]: Can this `if` give different results?

File: protocol/communities/community.go (hunk 1 above)
Comment (emizzle) [reply protocol/communities/community.go]: Thanks, I didn't realise we had this convenience method. Updated.

File: protocol/communities/community.go (hunk 2 above)
Comment (emizzle) [reply protocol/communities/community.go]: Of course, great look! Updated.

**Review by vkjr — APPROVED** [review]

**Comment by cammellos** [comment]
@emizzle tests are failing, I think it's the linting, you need to run:

```
goimports -w -local 'github.com/ethereum/go-ethereum,github.com/status-im/status-go,github.com/status-im/markdown' {filename}
```
https://ci.status.im/blue/organizations/jenkins/status-go%2Fprs%2Ftests/detail/PR-2236/3/pipeline

and it should fix it (you can set up your editor to do that automatically on save)

**Review by cammellos — APPROVED** [review]

**Comment by vkjr** [comment]
Just noticed that description not saved to database. Need to be fixed. Will add this.

File: protocol/communities/community_test.go [hunk 3]
```diff
@@ -143,6 +143,60 @@ func (s *CommunitySuite) TestCreateChat() {
...
+	s.Require().Equal(ErrNotAdmin, err)
+	
+	description := org.config.CommunityDescription
+	org.config.PrivateKey = s.identity
+	editChanges, err := org.EditChat(newChatID, &protobuf.CommunityChat{
+		Identity:    editedIdentity,
+		Permissions: editedPermissions,
+	})
+
+	s.Require().NoError(err)
```
Comment (vkjr) [inline protocol/communities/community_test.go:187]: Test should be updated to include work with database and ensuring chat description retainted after saving/loading

**Review by vkjr — CHANGES_REQUESTED** [review]
Description should be supported by database

**Comment by emizzle** [comment]
Thanks for the update @vkjr! Is this ready to go? If so, maybe the "changes requested" could be removed?

**Review by vkjr — APPROVED** [review]

**Comment by vkjr** [comment]
@emizzle, yeah, it is ready, removed request for change)

**Comment by vkjr** [comment]
@emizzle, fyi: I've rebased pr to `develop`

**Pull request merged**